creditcalc.py

```diff
@@ -9,36 +9,32 @@
 parser.add_argument("--payment", type=float)
 args = parser.parse_args()
 
-if args.interest is None:
-    print("Incorrect parameters.")
-    exit()
+if (args.type != "diff" and args.type != "annuity" or args.interest is None):
+    print("Incorrect parameters. --type can be 'diff' or 'annuity'. --interest is mandatory")
+    exit(-1)
 
-financial_data = [args.type, args.principal, args.periods, args.interest, args.payment]
 nominal_rate = args.interest / 12 / 100
 
-if args.type != 'diff' and args.type != 'annuity':
-    print("Incorrect parameters.")
-if len(financial_data) < 4:
-    print("Incorrect parameters.")
-if args.type == "diff" and args.payment is not None:
-    print("Incorrect parameters.")
-elif args.type == "diff":
+
+if args.type == "diff":
+    if args.principal is None or args.periods is None or args.payment is not None:
+        print("Missing required argument. Need to specify --principal and --periods, if using type='diff'.")
+        exit(-1)
     count = 0
     overpayment = 0
     for count in range(args.periods):
         count += 1
-        D = math.ceil(args.principal / args.periods + nominal_rate * (
-                    args.principal - (args.principal * (count - 1)) / args.periods))
-        overpayment = overpayment + D
-        print(f"Month {count}: payment is {D}")
+        installment = math.ceil(args.principal / args.periods + nominal_rate * (args.principal - (args.principal * (count - 1)) / args.periods))
```

Comment:
Unless `D` is some common financial notion, try to choose descriptive names for your variables.